src/Yoti.Auth/YotiClientEngine.cs

```diff
@@ -244,14 +256,25 @@ private static void HandleOtherAttributes(YotiUserProfile profile, AttrpubapiV1.
             }
         }
 
-        private string GetEndpoint(string token, string nonce, string timestamp, string sdkId)
+        private string GetEndpoint(HttpMethod httpMethod, string path, string token, string nonce, string timestamp, string sdkId, byte[] payload)
         {
-            return string.Format("/profile/{0}?nonce={1}&timestamp={2}&appId={3}", token, nonce, timestamp, sdkId);
+            string endpoint = string.Format("/{0}/{1}?nonce={2}&timestamp={3}&appId={4}", path, token, nonce, timestamp, sdkId);
+            if (payload != null && payload.Length > 0)
+            {
+                string urlSafePayloadString = Conversion.BytesToUrlSafeBase64(payload);
+                endpoint += string.Format("&payload={0}", urlSafePayloadString);
+            }
+
+            return endpoint;
```

Comment:
If this string is actually used as the url for the request, it will not work (although it accidentally works for GET requests).
This string should be used only as the input for the signed digest that you set as value of the "X-Yoti-Auth-Digest" header.

The url should contain only what you have up to line 261

Reply:
Ok sure, and the payload/content should be added as request.Content in [HttpRequester](https://github.com/getyoti/yoti-dotnet-sdk/pull/38/files#diff-26f682868312f987ad359165eadb85aa)?

Reply:
Yes, as it was supposed to be there (not if it is a GET obviously)